core/src/main/scala/org/apache/spark/deploy/rest/RestClient.scala

```diff
@@ -219,14 +221,20 @@ private[deploy] class StandaloneRestClient extends Logging {
 
   /** Return the base URL for communicating with the server, including the protocol version. */
   private def getBaseUrl(master: String): String = {
-    val masterUrl = master.stripPrefix("spark://").stripSuffix("/")
+    val masterUrl = if (master.startsWith("spark://")) {
+      master.stripPrefix("spark://").stripSuffix("/")
+    } else {
+      master.stripPrefix("mesos://").stripSuffix("/")
+    }
```

Comment:
This logic is duplicated here and in Spark submit. Maybe it makes sense to have
```
private val supportedMasterPrefixes = Seq("spark://", "mesos://")

private def getBaseUrl(master: String): String = {
  var masterUrl = master
  supportedMasterPrefixes.foreach { prefix =>
    if (master.startsWith(prefix)) {
      masterUrl = master.stripPrefix(prefix)
    }
  }
  masterUrl = masterUrl.stripSuffix("/")
  s"http://$masterUrl/..."
}

def validateMaster(master: String): Unit = {
  val valid = supportedMasterPrefixes.exists { prefix => master.startsWith(prefix) }
  if (!valid) {
    throw new IllegalArgumentException(
       "This REST client only supports master URLs that start with " +
       "one of the following: " + supportedMasterPrefixes.mkString(","))
  }
}
```
Then Spark submit can just call `validateMaster`. For that you might need to make this method static.